generated/nidcpower/nidcpower/_converters.py

```diff
@@ -149,6 +149,81 @@ def convert_repeated_capabilities_without_prefix(repeated_capability):
     return ','.join(convert_repeated_capabilities(repeated_capability, ''))
 
 
+def convert_single_group_repeated_capabilities(repeated_capability):
+    '''Convert a repeated capabilities string, with at most one instrument prefix (that ends with '/') and without any comma, to a list
+
+    Examples:
+        - '0' --> ['0']
+        - '0-2' --> ['0', '1', '2']
+        - '0:2' --> ['0', '1', '2']
+        - 'Dev1/0' --> ['Dev1/0']
+        - 'Dev1/0-2' --> ['Dev1/0', 'Dev1/1', 'Dev1/2']
+        - 'Dev1/0:2' --> ['Dev1/0', 'Dev1/1', 'Dev1/2']
+
+    Args:
+        repeated_capability (str) - Supported types (with an optional instrument prefix):
+            - range (using '-' or ':')
+            - single item
+
+    Returns:
+        rep_cap (str) - comma delimited list of strings of the expanded repeated capability items
+    '''
+    if '/' in repeated_capability:
+        split_index = repeated_capability.find('/') + 1
+        instrument_prefix = repeated_capability[:split_index]
+        return convert_repeated_capabilities(repeated_capability[split_index:], instrument_prefix)
+    return convert_repeated_capabilities(repeated_capability)
+
+
+def convert_channels_repeated_capabilities(
+    channels_repeated_capability,
+    first_channel_name
```

Comment:
> `first_channel_name` is confusing here since the function only needs the instrument name.

`first_channel_name` is not only used to extract the instrument name prefix, it is also used to determine whether the session is an independent-channels session (only independent-channels session would return `first_channel_name` with instrument prefix). I could rename it to `example_channel_name` if you prefer.

> But the more important point is, you need the instrument name because you're embedding FQCNs (Fully-qualified channel name like `dev1/4`) in the returned struct/namedtuple, in the case of single-instrument sessions.

Not really, I'm embedding FQCNs in the case of **independent-channels** sessions, which may be single-instrument or multi-instrument. This is also why I need `first_channel_name` to determine whether the session is an independent-channels session as mentioned earlier.

> I think you decided to embed FQCNs in single-instrument sessions because you need to take care of mixing i.e. an input like 4,Dev1/1:2'.

Nope, I'm embedding FQCNs in single-instrument session only when the session is an independent-channels session. In fact, non-FQCNs will be embedded in single-instrument sessions that are non-independent-channels:

```
>>> with nidcpower.Session(resource_name="dev1", channels="", options="Simulate=1, DriverSetup=Model:4162; BoardType:PXIe", independent_channels=False) as smu:
...     with smu.initiate():
...         smu.channels["4,1:2"].measure_multiple()
...
[Measurement(voltage=0.0, current=1e-05, in_compliance=None, channel='4'), Measurement(voltage=0.0, current=1e-05, in_compliance=None, channel='1'), Measurement(voltage=0.0, current=1e-05, in_compliance=None, channel='2')]
```

This **decision** was made based on the convention of the existing `get_channel_names()`, where **FQCNs are returned for independent-channels session** (regardless of whether it is single-instrument or multi-instrument) and **non-FQCNs are returned for non-independent-channels session** (this will always be single-instrument).

![image](https://user-images.githubusercontent.com/80089084/177119556-5e70b074-05b5-445b-a02a-b9d013726d59.png)

@sbethur @marcoskirsch With these being said, I believe the first 3 points of what you're proposing are not entirely valid since they are not consistent with the behavior of `get_channel_names()` in driver runtime. Point 4 is quite a different issue, and doing it would mean that it would no longer be consistent with the driver behavior. Nevertheless, I can file a GitHub issue for it if it's the desired behavior.

Reply:
After discussing with @marcoskirsch, it seems like "single-instrument sessions" and "multi-instrument sessions" mentioned in the original comment were actually terms used in NI-SCOPE that are equivalent to "non-independent-channels sessions" and "independent-channels sessions" in NI-DCPower.

> you need the instrument name because you're embedding FQCNs (Fully-qualified channel name like dev1/4) in the returned struct/namedtuple, in the case of single-instrument sessions.

Given the new interpretation of "single-instrument sessions", nope, I'm not embedding FQCNs in the case of single-instrument sessions / non-independent-channels sessions, as shown in the test cases of `test_fetch_multiple_channels()` where `independent_channels` is set to False and the `expected_measured_channel` is non-FQCN:
https://github.com/ni/nimi-python/pull/1754/files#diff-588aa0a2d97cf60012b577567bec4ae6bad13d2d7201507a924b8aa150bbe328R239

As for the proposed points, the first 2 points are already fulfilled, while the 3rd proposed point would never really occur, since user cannot pass in FQCNs for an NI-DCPower's "single-instrument session" / "non-independent-channels session":

```
>>> with nidcpower.Session(resource_name="Dev1", channels="", options="Simulate=1, DriverSetup=Model:4162; BoardType:PXIe", independent_channels=False) as smu:
...     smu.channels["Dev1/1"].voltage_level
...

nidcpower.errors.DriverError: -1074135008: IVI: Unknown channel or repeated capability name.

Channel Name: Dev1
```

With this, I believe all of the points have been addressed. I shall mark this conversation as resolved for now, feel free to reopen it if there's any further action item.